saleor/graphql/discount/mutations/voucher/voucher_add_catalogues.py

```diff
@@ -104,7 +104,9 @@ def perform_mutation(cls, _root, info: ResolveInfo, /, **data):
 
             if input_data:
                 manager = get_plugin_manager_promise(info.context).get()
-                cls.call_event(manager.voucher_updated, voucher, voucher.code)
+                code_instance = voucher.codes.last()
+                code = code_instance.code if code_instance else None
+                cls.call_event(manager.voucher_updated, voucher, code)
```

Comment:
I do not think it's any better than the previous solution. As we decided to keep `voucher.code` until 4.0, IMO changing it has no benefits.